hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -59,15 +69,15 @@
 
   public static final ConfigProperty<String> PLAN_STRATEGY_CLASS_NAME = ConfigProperty
       .key("hoodie.clustering.plan.strategy.class")
-      .defaultValue("org.apache.hudi.client.clustering.plan.strategy.SparkSizeBasedClusteringPlanStrategy")
+      .noDefaultValue()
```

Comment:
This will affect documentation. With this change, it would appear in the docs that Hudi is not setting any default value but in fact it is in the `build()` method. Either, we make sure that docs change is also done. 
Or, another way is to localize the change of defaults. Can it be done by overriding the `getWriteConfig()` method in `ClusteringPlanStrategy/ClusteringExecutionStrategy`? That way, we may not need `engineType` setting as well.

Reply:
Looks like I don't have to make this change to have engine-specific configs here.  I reverted the changes.  Regarding overriding `getWriteConfig()`, I think it may be better to put all configs in this class so it's easier to make changes in one place without confusion.

For the docs, do we need to mention that for Java engine, the default class names are different?

Reply:
Docs will be updated separately.